cluster/sync.go

```diff
@@ -2,37 +2,35 @@ package cluster
 
 import (
 	"strings"
+
+	"github.com/weaveworks/flux/resource"
 )
 
 // Definitions for use in synchronising a cluster with a git repo.
 
-// Yep, resources are defined by opaque bytes. It's up to the cluster
-// at the other end to do the right thing.
-type ResourceDef []byte
-
-// The action(s) to take on a particular resource.
-// This should just be done in order, i.e.,:
-//  1. delete if something in Delete
-//  2. apply if something in Apply
+// SyncAction represents either the deletion or application (create or
+// update) of a resource.
 type SyncAction struct {
-	// The ID is just a handle for labeling any error. No other
-	// meaning is attached to it.
-	ResourceID string
-	Delete     ResourceDef
-	Apply      ResourceDef
+	Delete     resource.Resource // ) one of these
+	Apply      resource.Resource // )
 }
 
 type SyncDef struct {
 	// The actions to undertake
 	Actions []SyncAction
 }
 
-type SyncError map[string]error
+type SyncError struct {
+	resource.Resource
+	Error error
+}
+
+type SyncErrors []SyncError
```

Comment:
Should this be its own commit? Could do with an explanation for what motivated the change, I'm sure future-us would agree.